### DISCUSSION TITLE
Compare API is not working. Urgent***

### DISCUSSION THREAD

**Discussion by Sri-Harsha-Gangavelli** [community/community#88457] [Actions]
### Select Topic Area

Bug

### Body

There is a known bug in **compare** api. I have more than 300 changed/added new files when I try to get the file changes from **comapre** api in **files** tag I am only getting 300 **files** at a time, even pagination is not working if I use pagination for the first page I am getting 300 **files** but for the second page **files** tag is missing. I guess pagination happening for the **commits** tag not for **files** tag. I need all **files** changes at a time or paginated would be great I am using in my backend application.

Attached screenshots for reference.

Please let me know the alternate approach!!.

![Screenshot-1](https://github.com/community/community/assets/155534767/91fccb9b-5ea4-49ac-9fad-d18c1e9e8ec8)
![Screenshot-2](https://github.com/community/community/assets/155534767/a18e342b-1d7a-4b8d-8993-4a4f8f71816b)

**Comment by shobrook** [comment]
Also experiencing this issue. I found that you can make the API return a response in the `diff` format by using the `"Accept: application/vnd.github.v3.diff"` header. This response _will_ include the names of all files that were changed, but it doesn't include the _type_ of change for each file (e.g. added/deleted/modified/renamed). This format also can't be easily parsed since its not a standard diff.
